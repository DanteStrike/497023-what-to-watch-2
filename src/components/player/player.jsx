import React from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";

import Video from "../video/video.jsx";
import {appActions, appSelectors} from "../../reducers/app/app";
import configs from "../../configs";


class Player extends React.PureComponent {
  constructor(props) {
    super(props);

    this._playerRef = React.createRef();
  }

  componentDidUpdate(prevState) {
    const {isFullScreen, toggleFullScreen} = this.props;
    const playerElement = this._playerRef.current;

    if (prevState.isFullScreen !== isFullScreen) {
      toggleFullScreen(playerElement);
    }
  }

  render() {
    const {
      isActivePlayer,
      videoSrc,
      updateProgressBar,
      renderProgressBar,
      renderPlayButton,
      renderFullScreen,
      closeVideoPlayer
    } = this.props;

    return (
      <div className="player" ref={this._playerRef}>
        <Video
          poster={configs.videoPlayerConfig.backgroundPoster}
          isActivePlayer={isActivePlayer}
          isMuted={true}
          src={videoSrc}
          isAutoReset={false}
          preload="metadata"
          updateProgressBar={updateProgressBar}
        />

        <button type="button" className="player__exit" onClick={closeVideoPlayer}>Exit</button>

        <div className="player__controls">
          <div className="player__controls-row">
            {renderProgressBar()}
          </div>

          <div className="player__controls-row">
            {renderPlayButton()}
            <div className="player__name">DanteStrike</div>
            {renderFullScreen()}
          </div>
        </div>
      </div>
    );
  }
}

Player.propTypes = {
  videoSrc: PropTypes.string.isRequired,
  isActivePlayer: PropTypes.bool.isRequired,
  isFullScreen: PropTypes.bool.isRequired,
  toggleFullScreen: PropTypes.func.isRequired,
  updateProgressBar: PropTypes.func.isRequired,
  renderProgressBar: PropTypes.func.isRequired,
  renderPlayButton: PropTypes.func.isRequired,
  renderFullScreen: PropTypes.func.isRequired,
  closeVideoPlayer: PropTypes.func.isRequired,
};

const mapStateToProps = (store) => ({
  videoSrc: appSelectors.getVideoPlayerInfo(store)
});

const mapDispatchToProps = (dispatch) => ({
  closeVideoPlayer: () => dispatch(appActions.closeVideoPlayer())
});

export {Player};
export default connect(mapStateToProps, mapDispatchToProps)(Player);
