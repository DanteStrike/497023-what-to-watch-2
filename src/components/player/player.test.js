import React from "react";
import renderer from "react-test-renderer";
import {Player} from "./player.jsx";


it(`Render correctly Player component`, () => {
  const component = renderer
    .create(
        <Player
          poster={`image`}
          src={`source`}
          preload={`none`}
          renderTimeBar={jest.fn()}
          renderPlayButton={jest.fn()}
          renderFullScreenButton={jest.fn()}
          closeVideoPlayer={jest.fn()}
          videoSrc={`url`}
        />
    ).toJSON();

  expect(component).toMatchSnapshot();
});
