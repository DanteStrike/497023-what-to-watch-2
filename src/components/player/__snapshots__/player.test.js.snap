// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render correctly Player component 1`] = `
<div
  className="player"
>
  <video
    className="player__video"
    poster="image"
    src="url"
  />
  <button
    className="player__exit"
    onClick={[MockFunction]}
    type="button"
  >
    Exit
  </button>
  <div
    className="player__controls"
  >
    <div
      className="player__controls-row"
    />
    <div
      className="player__controls-row"
    >
      <div
        className="player__name"
      >
        DanteStrike
      </div>
    </div>
  </div>
</div>
`;
